pandas/core/indexes/datetimelike.py

```diff
@@ -670,33 +670,29 @@ def intersection(self, other, sort=False):
         """
         self._validate_sort_keyword(sort)
         self._assert_can_do_setop(other)
-        res_name = get_op_result_name(self, other)
 
         if self.equals(other):
             return self._get_reconciled_name_object(other)
 
         if len(self) == 0:
-            return self.copy()
+            return self.copy()._get_reconciled_name_object(other)
         if len(other) == 0:
-            return other.copy()
+            return other.copy()._get_reconciled_name_object(self)
```

Comment:
in Index._union we dont copy, probably should

Reply:
The danger here is that someone accidentally modified the original version? Want me to add that to `Index._union` or save for later?

Reply:
lets go ahead and make it consistent now, yah